mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -811,19 +840,21 @@ public int executeBackendUpdateQuery(final BackendQuery<MongoDBQueryDescriptor>
 					backendQuery.getSingleEntityMetadataInformationOrNull().getEntityKeyMetadata();
 
 		String collectionName = getCollectionName( backendQuery, queryDescriptor, entityKeyMetadata );
-		DBCollection collection = provider.getDatabase().getCollection( collectionName );
+		MongoCollection<Document> collection = provider.getDatabase().getCollection( collectionName );
 
 		if ( !queryParameters.getPositionalParameters().isEmpty() ) { // TODO Implement binding positional parameters.
 			throw new UnsupportedOperationException( "Positional parameters are not yet supported for MongoDB native queries." );
 		}
 
 		switch ( queryDescriptor.getOperation() ) {
+			case INSERTMANY:
+			case INSERTONE:
 			case INSERT:
 				return doInsert( queryDescriptor, collection );
 			case REMOVE:
-				return doRemove( queryDescriptor, collection );
+				return (int) doRemove( queryDescriptor, collection );
```

Comment:
Well, what I mean is that this method is only used here, returning an int make more sense